t/plugin/proxy-mirror.t

```diff
@@ -445,3 +445,143 @@ GET /t
+                   }]]
+                   )
+
+               if code >= 300 then
+                   ngx.status = code
+               end
+               ngx.print(body)
+           }
+       }
+--- request
+GET /t
+--- error_code: 400
+--- response_body
+{"error_msg":"failed to check the configuration of plugin proxy-mirror err: property \"percentage\" validation failed: expected 10 to be smaller than 1"}
+--- no_error_log
+[error]
+
+
+
+=== TEST 14: set mirror requests percentage to 1
+--- config
+       location /t {
+           content_by_lua_block {
+               local t = require("lib.test_admin").test
+               local code, body = t('/apisix/admin/routes/1',
+                    ngx.HTTP_PUT,
+                    [[{
+                        "plugins": {
+                            "proxy-mirror": {
+                               "host": "http://127.0.0.1:1986",
+                               "percentage": 1
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                   }]]
+                   )
+
+               if code >= 300 then
+                   ngx.status = code
+               end
+               ngx.say(body)
+           }
+       }
+--- request
+GET /t
+--- error_code: 200
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 15: hit route with percentage 1
+--- request
+GET /hello?percentage=1
+--- error_code: 200
+--- response_body
+hello world
+--- error_log_like eval
+qr/uri: \/hello\?percentage=1/
+
+
+
+=== TEST 16: set mirror requests percentage to 0.5
+--- config
+       location /t {
+           content_by_lua_block {
+               local t = require("lib.test_admin").test
+               local code, body = t('/apisix/admin/routes/1',
+                    ngx.HTTP_PUT,
+                    [[{
+                        "plugins": {
+                            "proxy-mirror": {
+                               "host": "http://127.0.0.1:1986",
+                               "percentage": 0.5
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                   }]]
+                   )
+
+               if code >= 300 then
+                   ngx.status = code
+               end
+               ngx.say(body)
+           }
+       }
+--- request
+GET /t
+--- error_code: 200
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 17: hit route with percentage 0.5
```

Comment:
Could we check if almost 50% requests reach the mirror?

Reply:
Logs are currently checked~

50% it might not be very very accurate in the random case~

![image](https://user-images.githubusercontent.com/75366457/131965304-f12998c8-d564-46ea-bcb1-9ef8ca3c042f.png)

![image](https://user-images.githubusercontent.com/75366457/131965386-8a6d6c88-b5aa-47ca-98e8-346dfc596b52.png)

Reply:
What about trying it 180 times? Just like https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md#send-request

Reply:
TO design the almost range, I wrote a python script:
```
#!/usr/bin/env python3
# coding: utf-8
curr = [1, 1]
turns = 180
for i in range(1, turns+1):
    prev = curr
    curr = []
    curr.append(1)
    for j in range(1, i):
        curr.append(prev[j-1] + prev[j])
    curr.append(1)
n = sum(curr)
for i in range(turns+1):
    print("%d / %d is %f" % (i, turns, curr[i] / float(n)))
exp_per = 0.999
mid = int(turns / 2)
for i in range(1, mid):
    n_per = 0
    for j in range(mid - i, mid + i + 1):
        n_per += curr[j] / float(n)
        if n_per > exp_per:
            print(mid - i, j)
            break
    if n_per > exp_per:
        break

```

Reply:
okay~ thanks a lot~ done~